### PR TITLE
MEN-1775: make it possible to upload artifact to specific tenant 

### PR DESCRIPTION
**Pull request by GregorioDiStefano**
Signed-off-by: Gregorio Di Stefano <greg.distefano+github@gmail.com>

### CODE REVIEW

**Comment by GregorioDiStefano** [comment]
@maciejmrowiec this makes it possible to upload an artifact to a specific tenant, the code is essentially a copy of whats in the controller. 

Docs are missing, ill add them tomorrow morning. 

Does this need a unit test? I dont think it does, since it's essentially what we already have.

**Comment by maciejmrowiec** [comment]
@mendersoftware/rndity

**Comment by maciejmrowiec** [comment]
at least unit test on controller would be nice to have

**Comment by GregorioDiStefano** [comment]
@maciejmrowiec @kjaskiewiczz can someone take a look?
The test coverage went down because I refactored some reused code into utils/tests and its now counting as untested code (fixed by adding it the codecov ignore list, but needs to be merged for it to count..)

File: .codecov.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+codecov:
```
Comment (maciejmrowiec) [inline .codecov.yml]: will this append or overwrite main global config? (global config specify coverage policy)

Comment (maciejmrowiec) [reply .codecov.yml]: other trick is use `_test.go` files, tests don't count into golang coverage

**Comment by maciejmrowiec** [comment]
otherwhise looks ok

**Pull request merged**